lsp/src/Language/LSP/Server/Processing.hs

```diff
@@ -100,8 +101,9 @@ processMessage logger jsonStr = do
       FromClientMess m mess ->
         pure $ handle logger m mess
       FromClientRsp (P.Pair (ServerResponseCallback f) (Const !newMap)) res -> do
-        writeTVar pendingResponsesVar newMap
-        pure $ liftIO $ f (res ^. L.result)
+        unless shutdown <$> do
```

Comment:
```suggestion
        -- See Note [Shutdown]
        unless shutdown <$> do
```
and update the note to talk about responses

Reply:
I have made a flaws here, I think we should still update the pendingResponsesVar or we get into infinite loop proccessing the same response. Ajusted and add the doc.

Reply:
Why will we loop? AFAICT the only cost is that we will keep around response handler for _this_ response. We won't look at it again unless we get another response for that message, which we shouldn't do.

That said, it does seem neater to clean up the response handler and just not run it.